eyeshade/controllers/owners.js

```diff
@@ -154,46 +159,100 @@ const bulk = async (request, reply, runtime, owner, info, providers) => {
       verified: true,
       authorized: true,
       authority: owner,
       owner: owner,
-      visible: entry.show_verification_status || false,
       altcurrency: altcurrency,
-      info: info
-    })
+      info: underscore.defaults({ name: channel.authorizerName, email: channel.authorizerEmail }, info)
+    }, underscore.pick(props, [ 'providerName', 'providerSuffix', 'providerValue' ]))
 
-    await publishers.update({ publisher: entry.publisher }, state, { upsert: true })
+    await publishers.update({ publisher: channel.channelId }, state, { upsert: true })
 
-    entry.verificationId = uuid.v4().toLowerCase()
+    channel.verificationId = uuid.v4().toLowerCase()
     state.$set = underscore.extend(underscore.pick(state.$set, [ 'verified', 'visible' ]), {
-      token: entry.verificationId,
+      token: channel.verificationId,
       reason: 'bulk loaded',
       authority: owner,
       info: info
     })
-    await tokens.update({ publisher: entry.publisher, verificationId: entry.verificationId }, state, { upsert: true })
+    await tokens.update({ publisher: channel.channelId, verificationId: channel.verificationId }, state, { upsert: true })
 
     await runtime.queue.send(debug, 'publisher-report',
-                             underscore.extend({ owner: owner, publisher: entry.publisher },
+                             underscore.extend({ owner: owner, publisher: channel.channelId },
                                                underscore.pick(state.$set, [ 'verified', 'visible' ])))
   }
 
   reply({})
 }
 
+/*
+   DELETE /v1/owners/{owner}/{publisher}
+       [ used by publishers ]
+ */
+
+v1.deletePublisher = {
+  handler: (runtime) => {
+    return async (request, reply) => {
+      const owner = request.params.owner
+      const publisher = request.params.publisher
+      const debug = braveHapi.debug(module, request)
+      const owners = runtime.database.get('owners', debug)
+      const publishers = runtime.database.get('publishers', debug)
+      let entry, state
+
+      entry = await owners.findOne({ owner: owner })
+      if (!entry) return reply(boom.notFound('no such entry: ' + owner))
+
+      entry = await publishers.findOne({ owner: owner, publisher: publisher })
+      if (!entry) return reply(boom.notFound('no such entry: ' + publisher))
+
+      state = {
+        $currentDate: { timestamp: { $type: 'timestamp' } },
+        $unset: { owner: '' }
+      }
+      await publishers.update({ publisher: publisher }, state, { upsert: true })
+
+      reply({})
+    }
+  },
+
+  auth: {
+    strategy: 'simple',
+    mode: 'required'
+  },
+
+  description: 'Unlinks a publisher from an owner',
+  tags: [ 'api' ],
+
+  validate: {
+    params: {
+      owner: braveJoi.string().owner().required().description('the owner identity'),
+      publisher: braveJoi.string().publisher().required().description('the publisher identity')
+    },
+    query: {
+      access_token: Joi.string().guid().optional()
```

Comment:
is this used?

Reply:
it allows a person using swagger to test the endpoint!

Reply:
ah I see. that was not obvious to me since this is part of the query and not the authorization header as I would expect.

maybe we do it as a header validation instead? e.g.

```
validate: {
      headers: Joi.object({
        'authorization': Joi.string().required(),
      }).unknown()
}
```

I believe that would make it show up as an available field on swagger ala:
https://github.com/glennjones/hapi-swagger/issues/435